webapp/src/containers/addOrEditSubscriptions/subComponents/index.tsx

```diff
@@ -186,58 +131,46 @@ const AddOrEditSubscription = ({open, close, subscriptionData}: AddOrEditSubscri
-        handleError: (error) => {
-            if (error) {
-                if (error.id === Constants.ApiErrorIdRefreshTokenExpired || error.id === Constants.ApiErrorIdNotConnected) {
-                    dispatch(setConnected(false));
-                }
-
-                // We are not showing errors on the modal in case of these errors because we are creating an ephemeral post
-                if (
-                    error.id !== Constants.ApiErrorIdSubscriptionsNotConfigured &&
-                    error.id !== Constants.ApiErrorIdSubscriptionsUnauthorized &&
-                    error.id !== Constants.ApiErrorIdNotConnected
-                ) {
-                    setShowErrorComponent(true);
-                    setApiError(error);
-                    return;
-                }
-            }
-
-            dispatch(resetGlobalModalState());
-        },
-    });
-
     const {isLoading: createSubscriptionLoading} = getCreateSubscriptionState();
     const {isLoading: editSubscriptionLoading} = getEditSubscriptionState();
-    const {isLoading: subscriptionsConfiguredStateLoading,
-        isSuccess: subscriptionsConfiguredStateSuccess} = getSubscriptionsConfiguredState();
     const {isLoading: tableFieldsStateLoading, isError: tableFieldsStateError} = getTableFieldsState();
     const showLoader = createSubscriptionLoading || editSubscriptionLoading;
 
     useEffect(() => {
-        if (open && currentChannelId && !subscriptionData) {
-            setChannel(currentChannelId);
+        if (!open) {
+            return;
         }
 
-        if (open && subscriptionData && subscriptionsConfiguredStateSuccess) {
-            if (typeof (subscriptionData) === 'string') {
-                makeApiRequest(Constants.pluginApiServiceConfigs.fetchSubscription.apiServiceName, subscriptionData);
-            } else {
-                handleSubscriptionData(subscriptionData);
-                setEditing(true);
-            }
+        if (currentChannelId) {
+            setChannel(currentChannelId);
         }
-    }, [open, subscriptionData, subscriptionsConfiguredStateSuccess]);
 
-    useEffect(() => {
-        if (open && getTableFields) {
+        if (getTableFields) {
             makeApiRequest(Constants.pluginApiServiceConfigs.getTableFields.apiServiceName, Constants.SERVICENOW_SUBSCRIPTIONS_TABLE);
         }
-    }, [getTableFields]);
+
+        if (subscriptionData) {
+            // Set values for channel panel
+            setChannel(subscriptionData.channel);
+
+            // Set initial values for subscription-type panel
+            setSubscriptionType(subscriptionData.type);
+
+            // Set initial values for record-type panel
+            setRecordType(subscriptionData.recordType);
+
+            // Set initial values for search-record panel
+            setRecordId(subscriptionData.recordId);
+            setSuggestionChosen(true);
+
+            // Set initial value for filters panel
```

Comment:
maybe and same below as well ?
```suggestion
            // Set initial values for filters panel
```